File: VB/XpoXtraReports/Order.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.Xpo

Namespace XpoXtraReports

	Public Class Order
		Inherits XPObject
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
			' This constructor is used when an object is loaded from a persistent storage.
			' Do not place any code here.
		End Sub

		Public Overrides Sub AfterConstruction()
			MyBase.AfterConstruction()
			' Place here your initialization code.
			Me.Date = DateTime.Now
		End Sub

		#Region "Properties"
		Private _Date As DateTime
		Private _Id As Integer
		Public Property [Date]() As DateTime
			Get
				Return Me._Date
			End Get
			Set(ByVal value As DateTime)
				SetPropertyValue(Of DateTime)("Date", Me._Date, value)
			End Set
		End Property
		Public Property Id() As Integer
			Get
				Return Me._Id
			End Get
			Set(ByVal value As Integer)
				SetPropertyValue(Of Integer)("Id", Me._Id, value)
			End Set
		End Property
		#End Region 

		#Region "Property Lines (XPCollection)"
		<Association("Order-Lines", GetType(OrderLine)), Aggregated> _
		Public ReadOnly Property Lines() As XPCollection(Of OrderLine)
			Get
				Return GetCollection(Of OrderLine)("Lines")
			End Get
		End Property
		#End Region


	End Class
End Namespace

File: CS/XpoXtraReports/Order.cs
using System;
using DevExpress.Xpo;

namespace XpoXtraReports
{

    public class Order : XPObject
    {
        public Order(Session session)
            : base(session)
        {
            // This constructor is used when an object is loaded from a persistent storage.
            // Do not place any code here.
        }

        public override void AfterConstruction()
        {
            base.AfterConstruction();
            // Place here your initialization code.
            this.Date = DateTime.Now;
        }

        #region Properties
        private DateTime _Date;
        private int _Id;
        public DateTime Date{
            get { return this._Date; }
            set { SetPropertyValue<DateTime>("Date", ref this._Date, value); }
        }
        public int Id {
            get { return this._Id; }
            set { SetPropertyValue<int>("Id", ref this._Id, value); }
        }
        #endregion 

        #region Property Lines (XPCollection)
        [Association("Order-Lines", typeof(OrderLine)), Aggregated]
        public XPCollection<OrderLine> Lines
        {
            get { return GetCollection<OrderLine>("Lines"); }
        }
        #endregion
        
        
    }
}